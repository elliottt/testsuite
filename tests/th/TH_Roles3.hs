{-# LANGUAGE TemplateHaskell, PolyKinds, RoleAnnotations #-}

module Roles3 where

import Language.Haskell.TH

$( do { decls <- [d| data Foo a (b :: k) c@R (d :: k)@N |]
      ; reportWarning (pprint decls)
      ; return decls })
