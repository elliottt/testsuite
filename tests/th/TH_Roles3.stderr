
TH_Roles3.hs:7:4: Warning:
    data Foo_0 a_1 (b_2 :: k_3) c_4@R (d_5 :: k_3)@N
